databricks_cli/unity_catalog/delta_sharing_cli.py

```diff
@@ -171,6 +187,177 @@ def update_share_cli(api_client, name, new_name, comment, owner,
                       lambda json: UnityCatalogApi(api_client).update_share(name, json))
 
 
+def create_common_shared_data_object_options(f):
+    @click.option('--table', required=True,
+              help='Full name of the table to update from share.')
+    @click.option('--shared-as', default=None,
+                help='New name of the table inside the share.')
+    @click.option('--comment', default=None,
+                help='New comment of the table inside the share.')
+    @click.option('--partitions', default=None, type=JsonClickType(),
+                help='New partition specification of the table represented in JSON.')
+    @click.option('--cdf/--no-cdf', is_flag=True, default=None,
+                help='Enables change data feed of the table inside the share.')
+    @functools.wraps(f)
+    def wrapper(*args, **kwargs):
+        f(*args, **kwargs)
+    return wrapper
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Add a shared table.')
+@click.option('--share', required=True,
+              help='Name of the share to update.')
+@create_common_shared_data_object_options
+@click.option('--json-file', default=None, type=click.Path(),
+              help="Updates the shared table to shared data object represented in JSON file.")
+@click.option('--json', default=None, type=JsonClickType(),
+              help="Updates the shared table to shared data object represented in JSON.")
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def add_share_table_cli(api_client, share, table, shared_as, comment,
```

Comment:
In `update_share_cli`, you can set `hidden=true` for the add table and remove table options to encourage new usages to use these commands.